scanpy/tests/test_neighbors.py

```diff
@@ -114,41 +114,78 @@ def neigh() -> Neighbors:
     return get_neighbors()
 
 
+@pytest.mark.parametrize('method', ['umap', 'gauss'])
+def test_distances_euclidean(mocker, neigh, method):
+    """umap and gauss behave the same for distances.
+
+    They call pynndescent for large data.
+    """
+    from pynndescent import NNDescent
+
+    # When trying to compress a too-small index, pynndescent complains
+    mocker.patch.object(NNDescent, 'compress_index', return_val=None)
```

Comment:
What does this actually do? Are we just catching a warning?

Reply:
it crashes without that.